R/npn_get_species.R

```diff
@@ -40,74 +48,107 @@ npn_species_id <- function(ids, ...) {
+    kingdom <- rlang::arg_match0(kingdom, values = c("Plantae", "Animalia"))
+  }
+  req <- base_req %>%
+    httr2::req_url_path_append('species/getSpeciesByState.json') %>%
+    httr2::req_url_query(state = state, kingdom = kingdom)
+
+  resp <- httr2::req_perform(req)
+  out <- httr2::resp_body_json(resp, simplifyVector = TRUE)
+  #return:
+  tibble::as_tibble(out)
 }
 
 
 
 #' Species Search
 #'
-#' Search NPN species information using a number of different parameters, which can be used in conjunction with one another, including:
+#' Search NPN species information using a number of different parameters, which
+#' can be used in conjunction with one another, including:
 #'  - Species on which a particular group or groups are actually collecting data
 #'  - What species were observed in a given date range
 #'  - What species were observed at a particular station or stations
-#' @param network filter species based on a list of unique identifiers of NPN groups that are actually observing data on the species. Takes a list of IDs
-#' @param start_date filter species by date observed. This sets the start date of the date range and must be used in conjunction with end_date
-#' @param end_date filter species by date observed. This sets the end date of the date range and must be used in conjunction with start_date
-#' @param station_id filter species by a list of unique site identifiers
-#' @return A data frame with information on species in the NPN database filtered by partner group, dates and station/site IDs.
+#' @param network filter species based on identifiers of NPN groups that are
+#'   actually observing data on the species. Takes a single numeric ID.
+#' @param start_date filter species by date observed. This sets the start date
+#'   of the date range and must be used in conjunction with `end_date`.
+#' @param end_date filter species by date observed. This sets the end date of
+#'   the date range and must be used in conjunction with `start_date`.
+#' @param station_id filter species by a numeric vector of unique site
+#'   identifiers.
+#' @param ... Currently unused.
+#' @return A tibble with information on species in the NPN database filtered by
+#'   partner group, dates and station/site IDs.
 #' @export
 #' @rdname npn_species
-npn_species_search <- function(network=NULL, start_date=NULL, end_date=NULL, station_id=NULL, ...) {
-  args <- npnc(list(network_id = network, start_date = start_date,end_date = end_date))
-
-  for (i in seq_along(station_id)) {
-    args[paste0('station_ids[',i,']')] <- station_id[i]
-  }
-
-  ldfply(npn_GET(paste0(base(), 'species/getSpeciesFilter.json'), args, ...))
+npn_species_search <- function(network = NULL,
+                               start_date = NULL,
+                               end_date = NULL,
+                               station_id = NULL,
+                               ...) {
+  #TODO: clarify if multiple network IDs are allowed.  Can't test beause I don't
+  #know what IDs are valid
```

Comment:
Valid IDs are here: https://services.usanpn.org/npn_portal/networks/getNetworkTree.json

Keep behavior of current function with only 1 allowed, but make a note